api/routes/DoorCode.js

```diff
@@ -0,0 +1,119 @@
+const express = require('express');
+const router = express.Router();
+const DoorCode = require('../models/DoorCode');
+const {
+  checkIfTokenSent,
+  checkIfTokenValid,
+} = require('../util/token-functions');
+const {
+  OK,
+  BAD_REQUEST,
+  UNAUTHORIZED,
+  FORBIDDEN,
+  NOT_FOUND,
+} = require('../constants').STATUS_CODES;
+const addErrorLog = require('../util/logging-helpers');
+
+router.get('/getAvailableCodes', (req, res) => {
+  DoorCode.find({ usersAssigned: { $lt: 2 } }).then((doorcodes) =>
+    res.status(OK).send(doorcodes)
+  );
+});
+
+router.post('/addCode', (req, res) => {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(FORBIDDEN);
+  } else if (!checkIfTokenValid(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+
+  const newCode = new DoorCode({
+    doorCode: req.body.doorCode,
+    doorCodeValidUntil: req.body.doorCodeValidUntil,
+    usersAssigned: req.body.usersAssigned,
+  });
+
+  // Look for any repeated doorcodes
+  DoorCode.findOne(
```

Comment:
Instead of looking for repeated doorcodes inside the code, I realized you can add a unique field inside the schema itself. If you look in the User schema you can see that inside the email attribute they put a unique: true field. I believe this would be better and reduce code

Reply:
Also rebase after you make the changes